### PR TITLE
[BorrowPage] - replace borrow history endpoint with graphql

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.monday.com/boards/2218344956/pulses/2753866621

### CODE REVIEW

File: src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+          timestamp
+          transaction {
+            id
+          }
+        }
+      }
+    }`;
+  const { data, loading } = useQuery(BORROW_HISTORY);
+
+  return { data, loading };
```
Comment (creed-victor) [inline src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts:31]:
```suggestion
  return useQuery(BORROW_HISTORY);
```

File: src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+          interestDuration
+          collateralToLoanRate
+          timestamp
+          transaction {
+            id
+          }
+        }
+      }
+    }`;
+  const { data, loading } = useQuery(BORROW_HISTORY);
```
Comment (creed-victor) [inline src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts:29]: I think it should go outside of a hook, and maybe it should be not query but selector
And select borrows directly, not as a relation to user:

```suggestion
  const BORROW_HISTORY = gql`
    query getBorrowHistory ($user: ID!) {
        borrows (where: { user: $user } ) {
          loanId {
            id
          }
          loanToken
          collateralToken
          newPrincipal
          newCollateral
          interestRate
          interestDuration
          collateralToLoanRate
          timestamp
          transaction {
            id
          }
        }
    }`;
  const { data, loading } = useQuery(BORROW_HISTORY, { variables: { user: account.toLowerCase() }});
```

File: src/app/containers/BorrowHistory/index.tsx [hunk 3]
```diff
@@ -1,164 +1,76 @@
...
   const { t } = useTranslation();
+  const { data, loading } = useGetBorrowHistoryData();
 
-  const { events, loading } = useGetContractPastEvents(
-    'sovrynProtocol',
-    'Borrow',
-  );
+  const rows = useMemo(() => {
+    if (!data || loading) return null;
+    return data.user.borrows.map(item => {
```
Comment (creed-victor) [inline src/app/containers/BorrowHistory/index.tsx:17]: Could use borrows directly when skipping parent relation to user:

```suggestion
    return data.borrows.map(item => {
```

**Review by creed-victor — CHANGES_REQUESTED** [review]

File: src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts (hunk 2 above)
Comment (pietro-maximoff) [reply src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts:29]: yes right, i tried to implement it with selecting borrows directly, but with no luck and if i change it to your suggestion i will get an error - Cannot read properties of undefined (reading 'borrows')

File: src/app/containers/BorrowHistory/index.tsx (hunk 3 above)
Comment (pietro-maximoff) [reply src/app/containers/BorrowHistory/index.tsx:17]: related to https://github.com/DistributedCollective/Sovryn-frontend/pull/2251#discussion_r894343210

File: src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts (hunk 1 above)
Comment (pietro-maximoff) [reply src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts:31]: done

File: src/app/containers/BorrowHistory/index.tsx (hunk 3 above)
Comment (pietro-maximoff) [reply src/app/containers/BorrowHistory/index.tsx:17]: done

File: src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts (hunk 2 above)
Comment (pietro-maximoff) [reply src/app/containers/BorrowHistory/hooks/useGetBorrowHistoryData.ts]: done

**Review by soulBit — APPROVED** [review]

**Review by creed-victor — APPROVED** [review]

**Review by Alitasovryn — APPROVED** [review]

**Pull request merged**